aptos-move/framework/src/extended_checks.rs

```diff
@@ -560,35 +560,39 @@ impl<'a> ExtendedChecker<'a> {
             if !self.has_attribute(fun, VIEW_FUN_ATTRIBUTE) {
                 continue;
             }
-            self.check_transaction_args(&fun.get_id_loc(), &fun.get_parameter_types());
+            self.check_transaction_args(&fun.get_id_loc(), &fun.get_parameters());
             if fun.get_return_count() == 0 {
                 self.env
-                    .error(&fun.get_id_loc(), "view function must return values")
+                    .error(&fun.get_id_loc(), "`#[view]` function must return values")
             }
 
-            fun.get_parameter_types()
+            fun.get_parameters()
                 .iter()
-                .for_each(|parameter_type| match parameter_type {
-                    Type::Primitive(inner) => {
-                        if inner == &PrimitiveType::Signer {
-                            self.env.error(
-                                &fun.get_id_loc(),
-                                "view function cannot use the signer paremter",
-                            )
-                        }
-                    },
-                    Type::Reference(_, inner) => {
-                        if let Type::Primitive(inner) = inner.as_ref() {
+                .for_each(
+                    |Parameter(_sym, parameter_type, param_loc)| match parameter_type {
+                        Type::Primitive(inner) => {
                             if inner == &PrimitiveType::Signer {
                                 self.env.error(
-                                    &fun.get_id_loc(),
-                                    "view function cannot use the & signer paremter",
+                                    param_loc,
+                                    "`#[view]` function cannot use the `signer` parameter",
                                 )
                             }
-                        }
+                        },
+                        Type::Reference(mutability, inner) => {
+                            if let Type::Primitive(inner) = inner.as_ref() {
+                                if inner == &PrimitiveType::Signer
+                                    && mutability == &ReferenceKind::Immutable
```

Comment:
Any context on the reason why we need to check mutability here?

Reply:
added a comment.